File: src/main/java/com/hedvig/claims/commands/AddInitiatedAutomaticPaymentCommand.java
package com.hedvig.claims.commands;

import com.hedvig.claims.serviceIntegration.paymentService.dto.TransactionStatus;
import java.util.UUID;
import lombok.Value;
import org.axonframework.commandhandling.TargetAggregateIdentifier;

@Value
public class AddInitiatedAutomaticPaymentCommand {

  private String id;
  @TargetAggregateIdentifier
  private String claimId;
  private String memberId;
  private UUID transactionReference;
  private TransactionStatus transactionStatus;
}

File: src/main/java/com/hedvig/claims/commands/AddInitiatedAutomaticPaymentCommand.kt
package com.hedvig.claims.commands

import com.hedvig.claims.serviceIntegration.paymentService.dto.TransactionStatus
import org.axonframework.commandhandling.TargetAggregateIdentifier
import java.util.UUID

data class AddInitiatedAutomaticPaymentCommand(
    val id: String,
    @TargetAggregateIdentifier
    val claimId: String,
    val memberId: String,
    val transactionReference: UUID,
    val transactionStatus: TransactionStatus
)
